### PR TITLE
Orthotropy in MFront solid material models

### PR DESCRIPTION
**Pull request by endJunction**
Orthotropy in MFront solid material models 2D and 3D cases.
Tested against OGS orthotropic elasticity implementation.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.0)
2. [x] Tests covering your feature were added?
3. [ ] Any new feature or behavior change was documented?

### CODE REVIEW

File: MaterialLib/SolidModels/MFront/MFront.cpp [hunk 1]
```diff
@@ -293,19 +291,44 @@ MFront<DisplacementDim>::integrateStress(
...
+        }
+        return fourthOrderRotationMatrix(
+            _local_coordinate_system->transformation<DisplacementDim>(x));
+    }();
+
+    auto const eps_prev_MFront =
+        OGSToMFront(Q.transpose()
+                        .template topLeftCorner<
+                            KelvinVectorDimensions<DisplacementDim>::value,
+                            KelvinVectorDimensions<DisplacementDim>::value>() *
```
Comment (nagelt) [inline MaterialLib/SolidModels/MFront/MFront.cpp]: is the topLeftCorner stuff needed now that we have a 2D (4x4) variant of Q?

File: MaterialLib/SolidModels/MFront/MFront.cpp [hunk 2]
```diff
@@ -324,7 +347,10 @@ MFront<DisplacementDim>::integrateStress(
     {
         sigma[i] = behaviour_data.s1.thermodynamic_forces[i];
     }
-    sigma = MFrontToOGS(sigma);
+    sigma = Q.template topLeftCorner<
+                KelvinVectorDimensions<DisplacementDim>::value,
+                KelvinVectorDimensions<DisplacementDim>::value>() *
+            MFrontToOGS(sigma);
```
Comment (nagelt) [inline MaterialLib/SolidModels/MFront/MFront.cpp]: We rotate sigma back into OGS coordinate system, so epsilon and sigma will be output in xyz. All MFront internal variables (ElasticStrain for example) will be output in the local coordinate system. in the _behaviour.symmetry != mgis::behaviour::Behaviour::Symmetry::ISOTROPIC case we could output a warning that his is the case and users have to be careful. Having the local coordinate system as an output in ParaView for use in transformations would be convenient (but not necessary).

File: MaterialLib/SolidModels/MFront/MFront.cpp (hunk 1 above)
Comment (endJunction) [reply MaterialLib/SolidModels/MFront/MFront.cpp]: To be brief, yes.

File: MaterialLib/SolidModels/MFront/MFront.cpp (hunk 2 above)
Comment (endJunction) [reply MaterialLib/SolidModels/MFront/MFront.cpp]: A todo at the place where the rotation should happen added.

**Pull request merged**